src/App.tsx

```diff
@@ -1,223 +1,172 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+  FormEvent,
+} from 'react';
+
+import {
+  addTodos,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+} from './api/todos';
+
 import { AuthContext } from './components/Auth/AuthContext';
+import { TodosList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorNotification, ErrorText } from './components/ErrorNotification';
+import { FilterStatus } from './types/Filter';
+import { Todo } from './types/Todo';
+import { Header } from './components/Header';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterStatus, setFilterStatus] = useState('all');
+  const [title, setTitle] = useState('');
+  const [isAdding, setAdding] = useState(false);
+  const [loadingTodoIds] = useState<number[]>([]);
+  const [error, setError] = useState<ErrorText | null>(null);
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
-  useEffect(() => {
-    // focus the element with `ref={newTodoField}`
+  const filteredTodos = todos.filter(todo => {
+    switch (filterStatus) {
+      case FilterStatus.All:
+        return todo;
+
+      case FilterStatus.Active:
+        return !todo.completed;
+
+      case FilterStatus.Completed:
+        return todo.completed;
+
+      default:
+        throw new Error();
+    }
+  });
+
+  if (error) {
+    setTimeout(() => {
+      setError(null);
```

Comment:
I think if this setState changes Error it should be in the Error component. for now your error disappears instantly.